chain/client/src/view_client.rs

```diff
@@ -131,8 +137,32 @@ impl ViewClientActor {
                 }
             },
         };
+        let head = self.chain.head().map_err(|e| e.to_string())?.clone();
         if has_tx_result {
             let tx_result = self.chain.get_final_transaction_result(&tx_hash)?;
+            match tx_result.status {
+                FinalExecutionStatus::NotStarted | FinalExecutionStatus::Started => {
+                    for receipt_view in tx_result.receipts.iter() {
+                        let dst_shard_id = *self
+                            .chain
+                            .get_shard_id_for_receipt_id(&receipt_view.id)
+                            .map_err(|e| e.to_string())?;
+                        if self.chain.get_chunk_extra(&head.last_block_hash, dst_shard_id).is_err()
```

Comment:
Don't use `is_err` without checking that the error is `DbNotFound`

Reply:
Why is this a problem? It is fine if we send an extra request to some other peer.